internal/weldr/upload.go

```diff
@@ -137,7 +131,7 @@ func (u *uploadRequest) UnmarshalJSON(data []byte) error {
 	case "oci":
 		settings = new(ociUploadSettings)
 	case "generic.s3":
-		settings = new(genericS3UploadSettings)
+		settings = new(awsS3UploadSettings)
```

Comment:
I would like a comment in the code here explaining what's going on.

Reply:
I added a comment, please let me know if it's enough, or if you have any other suggestions